New documentation for News

Grotax opened the pull request:
This PR attempts to rework the current documentation.

Targets:
- recreate the documentation with mkdocs
- offer GitHub page for easy reading
- clean up of invalid or outdated documentation
- reorganization of the structure

SMillerDev commented:
Any way I can preview this?

Grotax commented:
> Any way I can preview this?

I guess only by building locally at this moment.
https://www.mkdocs.org/#installation

```
git checkout docs/mkdocs
mkdocs serve
```

Grotax commented:
I think I would delete the plugins documentation and probably we should deprecate the code.

Grotax commented:
> Any way I can preview this?

@SMillerDev for test purposes I have deployed the latest build here: https://grotax.de/

Grotax commented:
> The install.md should mention the 64-bit requirement.

yes true

> Also please add a newline in line 7 (before `You also need some PHP extensions:`) otherwise it's appended to the previous line in the generated html.
> 
> Furthermore there are multiple occurences of `*` in the html, which in fact should be bullet points (see install, faq, api-v1, api-v2).

yea I noticed that in some other files before apparently GitHub is more relaxed with markdown

Grotax commented:
After a manual deployment the page is now available on github https://nextcloud.github.io/news/

anoymouserver commented on docs/install.md:
```diff
@@ -1,6 +1,7 @@
+* 64bit OS (starting with News 16.0.0)
 * PHP >= 7.2
 * Nextcloud 20
 * libxml >= 2.7.8
```
```suggestion
* libxml >= 2.7.8

```
otherwise it is
- libxml >= 2.7.8 You also need some PHP extensions:

Grotax replied:
fixed

anoymouserver commented on docs/clients.md:
```diff
@@ -0,0 +1,31 @@
+| [FeedSpider](http://www.feedspider.net/)                                                                         | Firefox OS, webOS, LuneOS    | MIT License          | [GitHub](https://github.com/OthelloVentures/feedspider)            |
+| [fastReader](http://www.windowsphone.com/en-us/store/app/fastreader/e55e696d-aa45-4a49-bb1c-a1fc7fdabec1)        | Windows Phone                | -                    | -                                                                  |
+| [py3status](https://github.com/ultrabug/py3status/)                                                              | i3wm                         | BSD-3-Clause License | [GitHub](https://github.com/ultrabug/py3status/)                   |
+| [newsboat](http://newsboat.org/)                                                                                 | Unix Terminal                |                      | [GitHub](https://github.com/newsboat/newsboat)                     |
```
newsboat uses [MIT-License](https://github.com/newsboat/newsboat/blob/b506760bb61f46e8a5412cc67fa8d2d048229822/LICENSE)

Grotax replied:
fixed

anoymouserver commented on docs/clients.md:
```diff
@@ -0,0 +1,31 @@
+
+An update client uses the nextcloud API or the OCC CLI to update the feeds in News.
+
+* Python mulithread updater https://github.com/nextcloud/news-updater
```
```suggestion
* [Python multithread updater](https://github.com/nextcloud/news-updater)
```

But wasn't there an issue with this updater and the latest releases?

Grotax replied:
Well, as far as I know it still works with 15.4.x but I personally don't use it anymore.

I think we can leave it in there, when the repo gets fully abandoned we can remove it from the list or mark it as deprecated.

anoymouserver replied:
Ah, alright.

btw. `mul_t_ithread` 🙂

Grotax replied:
now fixed :D

anoymouserver reviewed (commented):
- We should maybe use up to date screenshots and not from 3rd parties - [FAQ](https://nextcloud.github.io/news/faq/#my-browser-shows-a-mixed-content-warning-connection-is-not-secure)
- More occurences of `*` instead of bullet points - [Installation/Update](https://nextcloud.github.io/news/install/), [FAQ](https://nextcloud.github.io/news/faq/)
- The link to the `background job file` is invalid - [FAQ](https://nextcloud.github.io/news/faq/#feeds-not-updated)

Grotax commented:
> * We should maybe use up to date screenshots and not from 3rd parties - [FAQ](https://nextcloud.github.io/news/faq/#my-browser-shows-a-mixed-content-warning-connection-is-not-secure)

Yea, I'm not sure about this whole section. I mean it was probably created due to bug reports by users.
Maybe it makes sense to do it less explicit and maybe reference the documentation of chrome and firefox (if there is any)?

> * More occurences of `*` instead of bullet points - [Installation/Update](https://nextcloud.github.io/news/install/), [FAQ](https://nextcloud.github.io/news/faq/)

should be fixed now

> * The link to the `background job file` is invalid - [FAQ](https://nextcloud.github.io/news/faq/#feeds-not-updated)
I updated the link and also replaced the references to the old docs in there

anoymouserver commented on docs/README.md:
```diff
@@ -0,0 +1,9 @@
+The News documentation of News uses mkdocs.
+As extension of mkdocs, mkdocs-material is used.
+
+Check the recourses below for how to install both of them locally.
```
```suggestion
Check the resources below for how to install both of them locally.
```

anoymouserver commented:
Maybe the docs link could also be updated here as well.
https://github.com/nextcloud/news/blob/271b6ee3c993264bd44a966b82f1a5daa85f53ab/appinfo/info.xml#L19-L22
(Not sure if there are any docs links in the app itself)

Grotax commented:
> Maybe the docs link could also be updated here as well.
> https://github.com/nextcloud/news/blob/271b6ee3c993264bd44a966b82f1a5daa85f53ab/appinfo/info.xml#L19-L22
> (Not sure if there are any docs links in the app itself)

Oh yea I forgot about those or rather didn't know how the URL would look like.

There are links in the admin page but I think I already changed those

Grotax commented:
I rebased and squashed everything, also fixed the last mistakes I could find.

I think it's ready to be merged, in another PR I would also adjust the stable15 branch:
- info.xml
- admin.php

The rest doesn't matter in my mind.
